gammapy/utils/fitting/base.py

```diff
@@ -0,0 +1,72 @@
+from __future__ import absolute_import, division, print_function, unicode_literals
+import logging
+import abc
+
+from astropy.utils.misc import InheritDocstrings
+
+from ...extern import six
+from .iminuit import fit_iminuit
+
+__all__ = ["Fit"]
+
+log = logging.getLogger(__name__)
+
+
+class FitMeta(InheritDocstrings, abc.ABCMeta):
+    pass
+
+
+@six.add_metaclass(FitMeta)
+class Fit(object):
+    """Abstract Fit base class.
+    """
+
+    @abc.abstractmethod
+    def total_stat(self, parameters):
```

Comment:
Does it make sense to pass `parameters` to `total_stat`?
Why not access them from `self._model.parameters` from this method?

Reply:
Right now `total_stat()` is directly used as the call-back function for the optimizers, so it is needed to pass in the parameters. For cleaner API I thought about making a private `_total_stat`, which is used as the call-back and then introducing a public property `.total_stat`.

Reply:
I like having ``total_stat`` as a public method accepting ``parameters``, because this is what is passed to the optimizers. Also it's useful to be able to calculated the Likelihood for a given set of model parameters without having to know about ``self._model`` e.g. to compute likelihood profiles